compiler/src/dotty/tools/dotc/core/Contexts.scala

```diff
@@ -266,25 +266,36 @@ object Contexts {
       base.sources.getOrElseUpdate(file, new SourceFile(file, codec))
     }
 
-    /** Sourcefile with given path name, memoized */
-    def getSource(path: TermName): SourceFile = base.sourceNamed.get(path) match {
+    /** SourceFile with given path name, memoized */
+    def getSource(path: TermName): SourceFile = base.sourceNamed.get(path) match
       case Some(source) =>
         source
-      case None => try {
-        val f = new PlainFile(Path(path.toString))
-        val src = getSource(f)
-        base.sourceNamed(path) = src
-        src
-      } catch {
-        case ex: InvalidPathException =>
-          report.error(s"invalid file path: ${ex.getMessage}")
-          NoSource
-      }
-    }
+      case None =>
+        getFile(path) match
+          case NoAbstractFile => NoSource
+          case file => getSource(file)
 
-    /** Sourcefile with given path, memoized */
+    /** SourceFile with given path, memoized */
     def getSource(path: String): SourceFile = getSource(path.toTermName)
 
+    /** AbstraFile with given path name, memoized */
```

Comment:
Is the memoization needed here? Does the Scala 2 compiler bridge do something similar?

Reply:
Following our private discussion I am trying to remove `sourceNamed` in https://github.com/lampepfl/dotty/pull/10607/commits/55e39801ecd0ff939431b382c691e7c99590836e